apps/s_cb.c

```diff
@@ -1355,3 +1356,46 @@ void ssl_ctx_security_debug(SSL_CTX *ctx, int verbose)
     SSL_CTX_set_security_callback(ctx, security_callback_debug);
     SSL_CTX_set0_security_ex_data(ctx, &sdb);
 }
+
+static void keylog_callback(const SSL *ssl, const char *line)
+{
+    /*
+     * There might be concurrent writers to the keylog file, so we must ensure
+     * that one line is written at once.
+     */
+    BIO_puts(bio_keylog, line);
+    BIO_write(bio_keylog, "\n", 1);
+    (void)BIO_flush(bio_keylog);
+}
```

Comment:
Is there any value in being cautious about the possibility of `bio_keylog` being `NULL` here? Maybe just adding an `assert` or something similar?